File: src/main/java/Class/Student.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Class;

/**
 *
 * @author huynh
 */
public class Student {

    private String studentId;
    private String name;
    private String email;
    private String dateOfBirth;
    private String gender;
    private String classId;
    private String academicYear;
    private String username;

    public Student() {

    }

    public Student(String studentId, String name, String email, String dateOfBirth, String gender, String classId, String academicYear) {
        this.studentId = studentId;
        this.name = name;
        this.email = email;
        this.dateOfBirth = dateOfBirth;
        this.gender = gender;
        this.classId = classId;
        this.academicYear = academicYear;
    }

    public String getStudentId() {
        return studentId;
    }

    public void setStudentId(String studentId) {
        this.studentId = studentId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getDateOfBirth() {
        return dateOfBirth;
    }

    public void setDateOfBirth(String dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getClassId() {
        return classId;
    }

    public void setClassId(String classId) {
        this.classId = classId;
    }

    public String getAcademicYear() {
        return academicYear;
    }

    public void setAcademicYear(String academicYear) {
        this.academicYear = academicYear;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

}

File: src/models/Student.js
/**
 * Student Model - OOP approach
 */
export class Student {
  constructor(
    studentId,
    name,
    email,
    dateOfBirth,
    gender,
    classId,
    academicYear,
    username = null
  ) {
    this.studentId = studentId;
    this.name = name;
    this.email = email;
    this.dateOfBirth = dateOfBirth;
    this.gender = gender;
    this.classId = classId;
    this.academicYear = academicYear;
    this.username = username;
  }

  getStudentId() {
    return this.studentId;
  }

  setStudentId(studentId) {
    this.studentId = studentId;
  }

  getName() {
    return this.name;
  }

  setName(name) {
    this.name = name;
  }

  getEmail() {
    return this.email;
  }

  setEmail(email) {
    this.email = email;
  }

  getDateOfBirth() {
    return this.dateOfBirth;
  }

  setDateOfBirth(dateOfBirth) {
    this.dateOfBirth = dateOfBirth;
  }

  getGender() {
    return this.gender;
  }

  setGender(gender) {
    this.gender = gender;
  }

  getClassId() {
    return this.classId;
  }

  setClassId(classId) {
    this.classId = classId;
  }

  getAcademicYear() {
    return this.academicYear;
  }

  setAcademicYear(academicYear) {
    this.academicYear = academicYear;
  }

  getUsername() {
    return this.username;
  }

  setUsername(username) {
    this.username = username;
  }

  toJSON() {
    return {
      studentId: this.studentId,
      name: this.name,
      email: this.email,
      dateOfBirth: this.dateOfBirth,
      gender: this.gender,
      classId: this.classId,
      academicYear: this.academicYear,
      username: this.username,
    };
  }
}
